config/routes.rb

```diff
@@ -1,3 +1,21 @@
 Rails.application.routes.draw do
+  root 'pages#home'
+  get 'addDoctor', to: 'pages#home'
+  get 'appointments', to: 'pages#home'
+  get 'DetailsPage/:id', to: 'pages#home'
+  get 'doctors', to: 'pages#home'
+  get 'DetailsPage/:id/book_appointment', to: 'pages#home'
+  get '/doctor/appointment', to: 'pages#home'
+  devise_for :clients
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  namespace :api do
+    namespace :v1 do
+      
+      resources :doctors  
```

Comment:
Do you need all RESTful actions in your API. I think you can limit them to the ones that are implemented.